src/publish/s3.js

```diff
@@ -3,9 +3,22 @@ const winston = require("winston-color")
 const zlib = require("zlib")
 const config = require("../config")
 
-const S3 = new AWS.S3()
+let conf = {}
 
+// This is the case where using custom s3 api-like services like minio.
+if (config.aws.endpoint) {
+  conf = {
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    endpoint: (config.aws.endpoint ? config.aws.endpoint : ''),
+    s3ForcePathStyle: 'true',
+    signatureVersion: 'v4'
+  }
```

Comment:
My preference would be to have these options in an S3 prop on the `src/config` object. Then we could pass that into the constructor. That way we could could configure these sorts of things in env vars, or let others change up the config on their fork.

I'm not sure that `s3ForcePathStyle` and `signatureVersion` need to be set to `true` and `v4` for all custom S3 endpoints, so I'm anxious about locking them down.